profiling/src/profile/mod.rs

```diff
@@ -431,6 +435,8 @@ impl Profile {
         self.endpoints
             .mappings
             .insert(interned_span_id, interned_endpoint);
+
+        self.endpoints.stats.add_endpoint(endpoint.to_string())
     }
```

Comment:
Hum ok, I was under the impression that `profiling endpoint` and reporting the statistics of profiled endpoints were tightly coupled.

For .NET, if profiling endpoints is deactivated, we won't be able to provide the counts. But if other (like Ruby) can and will different both, I can split this into 2 functions.

> Am I getting it right? Because... this may be slightly problematic, at least for Ruby, I think. Which I don't necessarily see as a blocker, since I can opt out. Here's the issue I'm seeing: this design assumes that add_endpoint will be called at most once for a given <local_root_span_id, endpoint> pair.

You are right, my assumption (at least for .NET), is that `add_endpoint` is called once. Generally, it's called when the trace is closed (when we know the endpoint name)

But I agree, it's  be good to dissociate those 2 behaviors.

Reply:
> Hum ok, I was under the impression that profiling endpoint and reporting the statistics of profiled endpoints were tightly coupled.

Not quite, as far as I understand it. Maybe @mar-kolya can chime in as well, but my understanding is that the statistics are supposed to represent what the tracer is seeing, not the profiling.

I believe the objective here is if in the space of 1 minute, the tracer sees 1000 traces on a given endpoint, and the profiler while sampling only sees 100 of those traces, we still want this counter to be 1000. (I'm not sure what the expected semantics are for traces that start before the 1-minute profile window, or those that extend beyond it).

Reply:
Hmm... Let me try to answer this to the best of my understanding.

@gleocadie Had asked me: should profiler provide these counts if endpoint profiler is disabled. And the answer to this - it largely doesn't matter. It is fine to have them but as things stand currently they will not be used.

But it looks like the question that is being discussed here may be different: should these counts include 'all' endpoint calls or only ones that are seen by the profiler - and the answer here is that these counts should include *all* andpoint calls. Even ones never sampled by the profiler. And even ones destined to be dropped by the tracer.

Does this make sense?

Reply:
Yeah that makes sense. Thx.